src/datastation/scripts/open_access_archeodepot.py

```diff
@@ -0,0 +1,150 @@
+from datastation.config import init
+from datastation.dv_api import publish_dataset, get_dataset_metadata, change_access_request, replace_dataset_metadata, \
+    change_file_restrict
+
+
+def open_access_archeodepot(datasets_file, licenses_file, dag_raporten_file, dataverse_config, dry_run, delay):
+    doi_to_license_uri = read_doi_to_license(datasets_file, read_rights_holder_to_license(licenses_file))
+    doi_to_dag_raporten = read_doi_to_dag_raporten(dag_raporten_file)
+    server_url = dataverse_config['server_url']
+    api_token = dataverse_config['api_token']
+    logging.debug("is dry run: {}".format(dry_run))
+    batch_process(doi_to_license_uri.items(),
+                  lambda key_value: update("doi:" + key_value[0],
+                                           key_value[1],
+                                           doi_to_dag_raporten.get(to_key(key_value[0]), []),
+                                           server_url,
+                                           api_token,
+                                           dry_run),
+                  delay)
+
+
+def read_doi_to_license(datasets_file, rights_holder_to_license_uri):
+    doi_to_license_uri = {}
+    with open(datasets_file, "r") as input_file_handler:
+        csv_reader = csv.DictReader(input_file_handler, delimiter=',', fieldnames=["DOI"], restkey="rest")
+        for row in csv_reader:
+            key = to_key(row["rest"][-1].strip())
+            uri = rights_holder_to_license_uri.get(key, "")
+            if uri:
+                doi_to_license_uri[row["DOI"]] = uri
+            else:
+                logging.warning("no license for line {}: {}".format(csv_reader.line_num, row))
+    return doi_to_license_uri
+
+
+def read_doi_to_dag_raporten(dag_raporten_file):
+    doi_to_dag_raporten = {}
+    with open(dag_raporten_file, "r") as input_file_handler:
+        csv_reader = csv.DictReader(input_file_handler, delimiter=',',
+                                    fieldnames=["dataset_id", "DOI"], restkey="files")
+        next(csv_reader)
+        for row in csv_reader:
+            doi_to_dag_raporten[to_key(row["DOI"])] = list(filter(lambda item: item != "", row["files"]))
+    return doi_to_dag_raporten
+
+
+def read_rights_holder_to_license(licenses_file):
+    rights_holder_to_license_uri = {}
+    with open(licenses_file, "r") as input_file_handler:
+        csv_reader = csv.DictReader(input_file_handler, delimiter=',')
+        for row in csv_reader:
+            rights_holder_to_license_uri[to_key(row["RIGHTS_HOLDER"])] = row["URI"]
+    return rights_holder_to_license_uri
+
+
+def to_key(name):
+    return re.sub("[^a-zA-Z0-1]", "_", name)
+
+
+def update(doi, uri, dag_raporten, server_url, api_token, dry_run):
+    resp_data = get_dataset_metadata(server_url, api_token, doi)
+    dirty = False
+    if resp_data['license']['uri'] != uri:
+        dirty = True
+        json_data = json.dumps({"http://schema.org/license": uri})
+        logging.info(json_data)
```

Comment:
i'd rather have a log-line saying something like:
`<doi> from <old-license> to <new-license>`

Reply:
Currently we have

```
2023-02-28 02:58:02 : INFO : batch_process : [1 of 1] Processing dataset with pid: ('10.5072/DAR/CCRDM7', 'http://creativecommons.org/licenses/by-nd/4.0')
2023-02-28 02:58:02 : INFO : update : {"http://schema.org/license": "http://creativecommons.org/licenses/by-nd/4.0"}```